### ISSUE TITLE
User role based visibility for UI elements

### ISSUE THREAD

**Issue by srivathsanvlb** [angular/angular#34248]
# 🚀 feature request

### Relevant Package

 This feature request is for @angular/core

### Description
 User role based visibility for UI elements

### Describe the solution you'd like
 We should have an inbuilt service that allows to capture the user role from the server and then have an attribute that can control visibility of UI elements.

### Describe alternatives you've considered
 ngx-permissions is widely is use. The feature can be built in to Angular.

**Comment by pkozlowski-opensource** [comment]
@srivathsanvlb thnx for the suggestion but I think that we will pass on it at this point. There are just way too many roles / permissions systems to plug them into Angular's core. If you use already ngx-permissions and it works to you than by all means continue to use it.

tl;dr; this is something better to tackle in the user-land.
